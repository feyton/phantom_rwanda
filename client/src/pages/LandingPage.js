import React from 'react';
import { useTranslation } from 'react-i18next';
import globe from '../assets/globe.svg';
import locator from '../assets/locator.png';
import start from '../assets/start.png';
import Card from '../components/Card.js';
import Intro from '../components/Intro.js';

const LandingPage = () => {
  const { t } = useTranslation();
  const cards = [
    {
      text: 'step_1_text',
      style: 'xl:justify-start xl:pt-2',
      image: start,
      alt: 'start',
      title: 'start'
    },
    {
      text: 'step_2_text',
      style: 'my-28 lg:my-0 xl:my-0 lg:mx-32 xl:mx-32',
      image: locator,
      alt: 'locator',
      title: 'step_2'
    },
    {
      text: 'step_3_text',
      style: '',
      image: globe,
      alt: 'globe',
      title: 'step_3'
    }
  ];
  return (
    <div>
      <div>
        <Intro />
        <div className="flex flex-col items-center">
          <h1 className="text-4xl font-raleway font-extrabold mt-28 xl:mt-44 mb-10 text-center">
            {t('how_it_work')}
          </h1>
          <div className=" flex flex-col lg:flex-row xl:flex-row mt-12 mb-32">
            {cards.map((card, index) => (
              <Card
                key={index}
                image={card.image}
                alt={card.alt}
                title={`${index + 1}. ${t(card.title)}`}
                text={t(card.text)}
                styles={card.style}
              />
            ))}
          </div>
        </div>
      </div>
    </div>
  );
};

export default LandingPage;
