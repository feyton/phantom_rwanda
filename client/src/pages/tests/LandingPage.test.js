import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import { MemoryRouter } from 'react-router-dom';
import LandingPage from '../LandingPage.js';

describe('Landing page test', () => {
  it('renders without crashing', () => {
    shallow(<LandingPage />);
  });
  it('should render LandingPage', () => {
    const elem = renderer
      .create(
        <MemoryRouter>
          <LandingPage />
        </MemoryRouter>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});
