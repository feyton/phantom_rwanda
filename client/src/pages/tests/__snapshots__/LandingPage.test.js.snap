// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page test should render LandingPage 1`] = `
<div>
  <div>
    <div
      className="flex flex-col items-center lg:flex-row xl:flex-row"
    >
      <div>
        <h1
          className="flex flex-col text-3xl md:text-5xl xl:text-5xl font-raleway font-extrabold mx-12 xl:mx-28 mt-24 md:mt-32 xl:mt-32"
        >
          <span>
            hero_head_1
          </span>
          <span
            className="md:ml-10 ml-4"
          >
            hero_head_2
          </span>
        </h1>
        <div
          className="flex flex-col text-sm md:text-xl xl:text-base mx-20 md:ml-32 xl:ml-52 mt-10"
        >
          <p>
            hero_text_1
          </p>
          <div>
            <p
              className="ml-4"
            >
              hero_text_2
            </p>
          </div>
        </div>
        <a
          href="/tracking-page"
          onClick={[Function]}
        >
          <div>
            <button
              className="bg-primary hover:bg-hover text-white rounded-3xl mt-12 px-12 xl:px-12 py-2 ml-28 md:ml-48 xl:ml-64 rounded-xl py-2 text-base md:text-sm px-5"
              type="submit"
            >
              hero_btn
            </button>
          </div>
        </a>
      </div>
      <div
        className="w-full px-20 lg:w-80 xl:w-96 lg:ml-auto xl:ml-auto lg:mr-36 xl:mr-36 lg:px-0 xl:px-0 mt-28"
      >
        <img
          alt="Bus"
          src="bus.png"
        />
      </div>
    </div>
    <div
      className="flex flex-col items-center"
    >
      <h1
        className="text-4xl font-raleway font-extrabold mt-28 xl:mt-44 mb-10 text-center"
      >
        how_it_work
      </h1>
      <div
        className=" flex flex-col lg:flex-row xl:flex-row mt-12 mb-32"
      >
        <div
          className="xl:justify-start xl:pt-2 border w-52 h-52 px-4 text-center flex flex-col justify-center items-center rounded-lg shadow-md"
        >
          <img
            alt="start"
            className="w-12 mt-4"
            src="start.png"
          />
          <h3
            className="font-raley mt-2 font-extrabold text-sm"
          >
            1. start
          </h3>
          <p
            className="font-sans text-center text-sm my-2"
          >
            step_1_text
          </p>
        </div>
        <div
          className="my-28 lg:my-0 xl:my-0 lg:mx-32 xl:mx-32 border w-52 h-52 px-4 text-center flex flex-col justify-center items-center rounded-lg shadow-md"
        >
          <img
            alt="locator"
            className="w-12 mt-4"
            src="locator.png"
          />
          <h3
            className="font-raley mt-2 font-extrabold text-sm"
          >
            2. step_2
          </h3>
          <p
            className="font-sans text-center text-sm my-2"
          >
            step_2_text
          </p>
        </div>
        <div
          className=" border w-52 h-52 px-4 text-center flex flex-col justify-center items-center rounded-lg shadow-md"
        >
          <img
            alt="globe"
            className="w-12 mt-4"
            src="globe.svg"
          />
          <h3
            className="font-raley mt-2 font-extrabold text-sm"
          >
            3. step_3
          </h3>
          <p
            className="font-sans text-center text-sm my-2"
          >
            step_3_text
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
